src/Forms/GridField/GridFieldAddExistingAutocompleter.php

```diff
@@ -106,6 +108,10 @@ public function getHTMLFragments($gridField)
     {
         $dataClass = $gridField->getModelClass();
 
+        if (!is_a($dataClass, DataObject::class, true)) {
```

Comment:
It's not really used in that many places.... I can do that if you want?

> I'm thinking it might be worthwhile to have an API on AbstractGridFieldComponent that explicitly define if a particular Gridfield compnonent works with non-DataObject.

What would that look like? An `init()` method or something? When would it be called, in the execution chain? I think that one requires a little more time in the oven.

Reply:
I thought that `AbstractGridFieldComponent` might be something useful instead of a pointless wrapper around the `GridFieldComponent` interface.

Reply:
It is already being useful - but I see what you mean. That can certainly be done, we would just need to consider what it would look like and do something with the fact that not all components will have it as a subclass yet. Perhaps in a major release we should make it the mandatory parent class of all components?